types/src/on_chain_config/gas_schedule.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::on_chain_config::OnChainConfig;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub struct GasSchedule {
+    pub entries: Vec<(String, u64)>,
+}
+
+impl GasSchedule {
+    pub fn to_btree_map(self) -> BTreeMap<String, u64> {
```

Comment:
How/when is this config reloaded from latest on chain state?

Reply:
When you create an AptosVM, it will try to load the config from the state view

Reply:
Then it sounds like it shouldn't need reconfiguration to see gas schedule updates? Anyway this can be figured out in a separate followup PR as part of making sure gas schedule is upgradable :)